lib/pages/bookmark/group/group_article_list_page.dart

```diff
@@ -101,42 +100,106 @@ class _GroupArticleListPageState extends State<GroupArticleListPage> {
+    List<String> tags = [];
+    List<String> series = [];
+    List<String> characters = [];
+    if (articleEh.female != null) {
+      articleEh.female?.forEach((female) {
+        tags.add('female:$female');
+      });
+    }
+    if (articleEh.male != null) {
+      articleEh.male?.forEach((male) {
+        tags.add('male:$male');
+      });
+    }
+    if (articleEh.misc != null) {
+      articleEh.misc?.forEach((misc) {
+        tags.add('tag:$misc');
+      });
+    }
+    if (articleEh.parody != null) {
+      articleEh.parody?.forEach((parody) {
+        series.add(parody);
+      });
+    }
+    if (articleEh.character != null) {
+      articleEh.character?.forEach((character) {
+        characters.add(character);
+      });
+    }
     final meta = {
       'Id': int.parse(id),
       'EHash': hash,
       'Title': articleEh.title,
       'Artists': articleEh.artist?.join('|') ?? 'N/A',
+      'Language': articleEh.languages?.join('|') ??
+          articleEh.language.trim().toLowerCase(),
+      'Tags': tags.join('|'),
+      'Characters': characters.join('|'),
+      'Type': articleEh.type,
+      'Series': series.join('|'),
     };
     return QueryResult(result: meta);
   }
 
   Future<QueryResult> _tryGetArticleFromExhentai(String id) async {
-    final listHtml = await EHSession.requestString(
-        'https://exhentai.org/?next=${(int.parse(id) + 1)}');
-    final href =
-        parse(listHtml).querySelector('a[href*="/g/$id/"]')?.attributes['href'];
-    final hash = href!.split('/').lastWhere((element) => element.isNotEmpty);
+    final hash = await EHSession.getEHashById(id, 'exhentai.org');
     final html = await EHSession.requestString(
         'https://exhentai.org/g/$id/$hash/?p=0&inline_set=ts_m');
     final articleEh = EHParser.parseArticleData(html);
+    List<String> tags = [];
+    List<String> series = [];
+    List<String> characters = [];
+    if (articleEh.female != null) {
+      articleEh.female?.forEach((female) {
+        tags.add('female:$female');
+      });
+    }
+    if (articleEh.male != null) {
+      articleEh.male?.forEach((male) {
+        tags.add('male:$male');
+      });
+    }
+    if (articleEh.misc != null) {
+      articleEh.misc?.forEach((misc) {
+        tags.add('tag:$misc');
+      });
+    }
+    if (articleEh.parody != null) {
+      articleEh.parody?.forEach((parody) {
+        series.add(parody);
+      });
+    }
+    if (articleEh.character != null) {
+      articleEh.character?.forEach((character) {
+        characters.add(character);
+      });
+    }
     final meta = {
       'Id': int.parse(id),
       'EHash': hash,
       'Title': articleEh.title,
       'Artists': articleEh.artist?.join('|') ?? 'N/A',
+      'Language': articleEh.languages?.join('|') ??
+          articleEh.language.trim().toLowerCase(),
+      'Tags': tags.join('|'),
+      'Characters': characters.join('|'),
+      'Type': articleEh.type,
+      'Series': series.join('|'),
     };
```

Comment:
```suggestion
    final article = EHParser.parseArticleData(html);
    final meta = {
      'Id': int.parse(id),
      'EHash': hash,
      'Title': article.title,
      'Artists': article.artist?.join('|') ?? 'N/A',
      'Language':
          article.languages?.join('|') ?? article.language.trim().toLowerCase(),
      'Tags': ((article.female?.map((e) => 'female:$e').toList() ?? []) +
              (article.male?.map((e) => 'male:$e').toList() ?? []) +
              (article.misc?.map((e) => 'tag:$e').toList() ?? []))
          .join('|'),
      'Characters': article.character?.join('|') ?? '',
      'Type': article.type,
      'Series': article.parody?.join('|') ?? '',
    };
```

QueryResult에 From EHArticle 메서드를 만들어도 좋을 것 같네요